rockchip: fix mmc core set initial signal voltage on reboot

peterwillcn opened the pull request:
Some boards have SD card connectors where the power rail cannot be switched
off by the driver. If the card has not been power cycled, it may still be
using 1.8V signaling after a warm re-boot. Bootroms expecting 3.3V signaling
will fail to boot from a UHS card that continue to use 1.8V signaling.

Set initial signal voltage in mmc_power_off() to allow re-boot to function.

This fixes re-boot with UHS cards on Rockchip RK3288, RK3399 boards.

original author Jonas Karlman <jonas@kwiboo.se>

AmadeusGhost commented:
https://patchwork.kernel.org/project/linux-mmc/patch/AM3PR03MB09664161A7FA2BD68B2800A7AC620@AM3PR03MB0966.eurprd03.prod.outlook.com/
https://github.com/immortalwrt/immortalwrt/blob/master/target/linux/rockchip/patches-5.4/105-mmc-core-set-initial-signal-voltage-on-power-off.patch
Why do you delete the original author's information and replace it into yourself?

peterwillcn committed:
rockchip: fix mmc core set initial signal voltage on reboot

Some boards have SD card connectors where the power rail cannot be switched
off by the driver. If the card has not been power cycled, it may still be
using 1.8V signaling after a warm re-boot. Bootroms expecting 3.3V signaling
will fail to boot from a UHS card that continue to use 1.8V signaling.

Set initial signal voltage in mmc_power_off() to allow re-boot to function.

This fixes re-boot with UHS cards on Rockchip RK3288, RK3399 boards.

Signed-off-by: xiaobo <peterwillcn@gmail.com>

peterwillcn commented:
@AmadeusGhost There is no politics in the code world, there is no infighting, it just happens that the fix method is the same, and it doesn't matter what author is added, fixing the bug to make everyone's device work properly is the most important. If you have a good fix please submit pr, I can offer you this opportunity.

AmadeusGhost commented:
Again, that is not the reason you did it: You keep the original author’s commit description and replace the author information and Signed-off-by fields for yourself. Don't try to avoid the problem, why can't you keep the original submission information of the patch? To be honest, your behavior is extremely disrespectful to the original author and the open source community.
BTW, it seems you have done this more than once: https://github.com/openwrt/openwrt/pull/3040#issuecomment-632812683

blocktrron commented:
Is this also called upon a kernel panic? Reading from the original commit description, if this is not the case the board will lock up.

Not that I have a better solution, I was just wondering about this circumstance.

> There is no politics in the code world

I'd suggest you rethink your position here, as politics and the attached systems made OpenWrt possible in the first place. But I've had enough to do with that already, so i will not explain this in detail here. @AmadeusGhost already brought up the most relevant points  IMHO.

peterwillcn commented:
@blocktrron Maybe there is a better solution. I want to see what I can do in the Uboot
@AmadeusGhost You're not the author who wants everyone to respect you. If you're not going to solve problems and help people, then shut up. [Are you promoting your new lede down the road? Maybe your branch has a back door to steal everyone's secrets](https://github.com/AmadeusGhost/lede)

AmadeusGhost commented:
So do you not think your behavior has problems? The plagiarism code is shameful.
Btw, my own fork code has nothing to do with you, there is no need to take the opportunity to change the subject.

peterwillcn commented:
@AmadeusGhost Can you tell that branches don't steal other people's code and it's all your own? Write your own code from scratch, don't use open source Linux, arrogant guy.

peterwillcn commented:
@AmadeusGhost Don't judge others from the moral high ground, Talk is cheap. Show me the code.

AmadeusGhost commented:
It seems that this person is changing the topic and  personal attacks, can we lock this?

peterwillcn commented:
@AmadeusGhost Where there is an attack on you, is to say things objectively, if you are the original author, respect you do not have any problem, the problem is that you are not, [if you can help us solve this problem](https://forum.openwrt.org/t/nanopi-r4s-rk3399-4g-is-a-great-new-openwrt-device/79143/268), if you can not solve then shut up, Talk is cheap. Show me the code.
